### PR TITLE
[WFLY-17961] Spurious errors from Micrometer

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-17961

Implement a no-op MeterRegistry
Make the endpoint not required
If the endpoint is not specified, register the no-op MeterRegistry

### CODE REVIEW

File: observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerExtensionLogger.java [hunk 1]
```diff
@@ -75,4 +75,8 @@ public interface MicrometerExtensionLogger extends BasicLogger {
     @LogMessage(level = INFO)
     @Message(id = 10, value = "Not activating Micrometer Subsystem")
     void notActivatingSubsystem();
+
+    @LogMessage(level = WARN)
+    @Message(id = 11, value = "Micrometer has been enabled, but no endpoint has been configured. A No-op registry has been configured.")
```
Comment (bstansberry) [inline observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerExtensionLogger.java:80]: s/registry/metrics registry/g ?

File: observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerRegistryService.java [hunk 2]
```diff
@@ -18,19 +18,22 @@
...
 import org.jboss.msc.service.StopContext;
 import org.wildfly.extension.micrometer.jmx.JmxMicrometerCollector;
-import org.wildfly.extension.micrometer.metrics.WildFlyRegistry;
+import org.wildfly.extension.micrometer.registry.NoOpRegistry;
+import org.wildfly.extension.micrometer.registry.WildFlyOtlpRegistry;
+import org.wildfly.extension.micrometer.registry.WildFlyRegistry;
+
+import java.io.IOException;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
```
Comment (bstansberry) [inline observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerRegistryService.java:36]: These were in the right spot before :)

File: observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemDefinition.java [hunk 3]
```diff
@@ -68,7 +68,7 @@ class MicrometerSubsystemDefinition extends PersistentResourceDefinition {
     public static final SimpleAttributeDefinition ENDPOINT = SimpleAttributeDefinitionBuilder
             .create(MicrometerConfigurationConstants.ENDPOINT, ModelType.STRING)
             .setAttributeGroup(MicrometerConfigurationConstants.OTLP_REGISTRY)
-            .setRequired(true)
+            .setRequired(false)
```
Comment (bstansberry) [inline observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemDefinition.java:71]: You need to bump the management API version for this. XSD as well.

File: observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerRegistryService.java (hunk 2 above)
Comment (jasondlee) [reply observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerRegistryService.java:36]: Grr. I had IDEA configured correctly at one point. :P

File: observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerExtensionLogger.java (hunk 1 above)
Comment (jasondlee) [reply observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerExtensionLogger.java:80]: Changed.

File: observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemDefinition.java (hunk 3 above)
Comment (bstansberry) [reply observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemDefinition.java:71]: Note that this subsystem is not supported in domain mode in WF 28, so there is no need to deal with mixed domain transformation.

**Review by darranl — CHANGES_REQUESTED** [review]
Please separate the model bump aspect into it's own commit.  Using current tooling it is not possible to see what has changed in the schema and in future using git annotate it will not be possible to see the first changes that were made to the schema as it all shows as a new file in a single commit.

**Review by bstansberry — APPROVED** [review]

**Review by darranl — APPROVED** [review]

**Comment by darranl** [comment]
Just removing ready for merge as it is suspected of being related to a test failure on the aggregate job.

**Comment by darranl** [comment]
Sorry @jasondlee this one has a merge conflict, also it is the Bootable Jar Linux run we saw failing in the aggregate job which was why this one didn't go ahead.

**Comment by jasondlee** [comment]
> Sorry @jasondlee this one has a merge conflict, also it is the Bootable Jar Linux run we saw failing in the aggregate job which was why this one didn't go ahead.

Not a problem. i'll get that resolved ASAP.

**Pull request merged**